packages/ens-utils/src/ethregistrar.ts

```diff
@@ -155,3 +179,284 @@ export function buildNFTRefFromENSName(
+): Price | null => {
+  if (registration.expirationTimestamp) {
+    return temporaryPremiumPriceAtTimestamp(
+      now(),
+      registration.expirationTimestamp,
+    );
+  } else {
+    return null;
+  }
+};
+
+/* Interface for premium period end details */
+export interface PremiumPeriodEndsIn {
+  relativeTimestamp: string;
+  timestamp: Timestamp;
+}
+
+/**
+ * Determines if a domain is in its premium period and returns the end timestamp and a human-readable distance to it.
+ * @param domainCard: DomainCard
+ * @returns PremiumPeriodEndsIn | null
+ */
+export const premiumPeriodEndsIn = (
+  registration: Registration,
+): PremiumPeriodEndsIn | null => {
+  const isExpired =
+    registration.primaryStatus === PrimaryRegistrationStatus.Expired;
+  const wasRecentlyReleased =
+    registration.secondaryStatus ===
+    SecondaryRegistrationStatus.RecentlyReleased;
+
+  /*
+    A domain will only have a premium price if it has Expired and it was Recently Released
+  */
+  if (!isExpired || !wasRecentlyReleased) return null;
+
+  /*
+    This conditional should always be true because expiryTimestamp will only be null when
+    the domain was never registered before. Considering that the domain is Expired,
+    it means that it was registered before. It is just a type safety check.
+  */
+  if (!registration.expiryTimestamp) return null;
+
+  const releasedEpoch = addSeconds(registration.expiryTimestamp, GRACE_PERIOD);
+  const temporaryPremiumEndTimestamp = addSeconds(
+    releasedEpoch,
+    TEMPORARY_PREMIUM_PERIOD,
+  );
+  console.log({
+    relativeTimestamp: formatTimestampAsDistanceToNow(
+      temporaryPremiumEndTimestamp,
+    ),
+    timestamp: temporaryPremiumEndTimestamp,
+  });
+  return {
+    relativeTimestamp: formatTimestampAsDistanceToNow(
+      temporaryPremiumEndTimestamp,
+    ),
+    timestamp: temporaryPremiumEndTimestamp,
+  };
+};
+
+/**
+ * At the moment a .eth name expires, this recently released temporary premium is added to its price.
+ * NOTE: The actual recently released temporary premium added subtracts `PREMIUM_OFFSET`.
+ */
+export const PREMIUM_START_PRICE: Price = {
+  value: 10000000000n /* $100,000,000.00 (100 million USD) */,
+  currency: Currency.Usd,
+};
+
+/**
+ * The recently released temporary premium drops exponentially by 50% each day.
+ */
+const PREMIUM_DECAY = 0.5;
+
+/**
+ * Goal:
+ *  The temporary premium should drop to $0.00 after exactly `PREMIUM_DAYS` days have passed.
+ *
+ * Challenge:
+ *  If we decay `PREMIUM_START` by a rate of `PREMIUM_DECAY` each day over the course of
+ *  `PREMIUM_DAYS` days we don't get $0.00 USD. Instead, we get this `PREMIUM_OFFSET` value
+ *  ($47.68 USD).
+ *
+ * Solution:
+ *  Subtract this value from the decayed temporary premium to get the actual temporary premium.
+ */
+export const PREMIUM_OFFSET = approxScalePrice(
+  PREMIUM_START_PRICE,
+  PREMIUM_DECAY ** Number(TEMPORARY_PREMIUM_DAYS),
+);
+
+export function temporaryPremiumPriceAtTimestamp(
```

Comment:
It will be awesome if you can create unit tests for this :+1:

Reply:
I have added the wrapper for the test scenarios that should exist for this function, the tests itself depend on Registration objects which will be available in the codebase once `mock-utils` is created. 

OK to wait for `mock-utils` and `getMockedRegistration` before implementing the test scenarios internal logics?